src/libraries/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -2097,46 +2117,30 @@ internal static class ReflectionXmlSerializationReaderHelper
 
         public static SetMemberValueDelegate GetSetMemberValueDelegateWithType<TObj, TParam>(MemberInfo memberInfo)
         {
-            if (typeof(TObj).IsValueType)
```

Comment:
Would it make sense to add `Debug.Assert(!typeof(TObj).IsValueType);`? Just to document that we don't expect any structs to come into this method.